### PR TITLE
Issue #3051857 by agami4: show background image after saving on settings theme page

### PR DESCRIPTION
**Pull request by agami4**
## Problem
Show Background Image switch settings not saving

## Solution
Show background image after saving on settings theme page

## Issue tracker
https://www.drupal.org/project/social/issues/3051857

## How to test
- [ ] Go to the settings theme page
- [ ] Click on switch button to show background banner image
- [ ] Save changes
- [ ] Reload page

## Release notes
When you click switch button to show background banner image and save changes, you can see this background banner image after reload page

### CODE REVIEW

File: themes/socialblue/color/preview.js [hunk 1]
```diff
@@ -64,10 +64,35 @@
...
-          .toggleClass("cover-img-gradient");
+      var localStorage =  window.localStorage;
+      var colorPreviewLever = $('.color-preview-hero .switch .lever');
+      var colorPreviewCover = $('.color-preview-hero .cover');
+
+      if (localStorage.getItem('lever-on') &&
+        localStorage.getItem('set-settings-theme')) {
+          colorPreviewLever.addClass('lever-on');
+          colorPreviewCover.addClass('cover-img')
+            .addClass('cover-img-gradient');
```
Comment (Kingdutch) [inline themes/socialblue/color/preview.js]: We can move this in with the previous `addClass` call, no need for two separate function calls.

File: themes/socialblue/color/preview.js [hunk 2]
```diff
@@ -64,10 +64,35 @@
...
+      }
+
+      $colorPreview.find('.color-preview-hero .switch').once()
+        .click(function() {
+          colorPreviewLever.toggleClass('lever-on');
+          colorPreviewCover.toggleClass('cover-img')
+            .toggleClass('cover-img-gradient');
```
Comment (Kingdutch) [inline themes/socialblue/color/preview.js]: We can move this in with the previous `toggleClass` call, no need for two separate function calls.

See http://api.jquery.com/toggleclass/
> Applying `.toggleClass( "bounce spin" )` to the same `<div>` alternates between `<div class="tumble bounce spin">` and `<div class="tumble">`.

File: themes/socialblue/color/preview.js [hunk 3]
```diff
@@ -64,10 +64,35 @@
...
+          }
+          else {
+            localStorage.removeItem('lever-on');
+          }
+        }
+      );
+
+      var settingThSub = $('.system-theme-settings .form-actions .form-submit');
+      settingThSub.on('click', function () {
+        localStorage.setItem('set-settings-theme', '1');
```
Comment (Kingdutch) [inline themes/socialblue/color/preview.js]: I think we can just remove this setting entirely and just persist the user's preference regardless of whether they actually saved. After the first time they've saved this page that will be the behaviour anyway.

**Comment by jaapjan** [comment]
@agami4 still some time to implement requested changes from @Kingdutch ?

File: themes/socialblue/color/preview.js (hunk 3 above)
Comment (agami4) [reply themes/socialblue/color/preview.js]: If I understood you right, you want save it without localStorage? save on color.inc file?

Comment (Kingdutch) [reply themes/socialblue/color/preview.js]: Sorry I didn't really word it clearly (had to re-figure it out myself as well). The localStorage is already updated when the user toggles the setting (see https://github.com/goalgorilla/open_social/pull/1356/files/67a698be713c89e7b4677b853a2f630ffa31b8ec#diff-f6039e2aa13ec5f04b7b67d2a88d39a5R84 ). 

There's no real reason to also do this when the user saves the form. So we can remove line 95 :)

**Comment by jaapjan** [comment]
Will be postponed to 6.1

**Comment by agami4** [comment]
@jaapjan I updated all comments from @Kingdutch

File: themes/socialblue/color/preview.js [hunk 4]
```diff
@@ -8,12 +8,14 @@
 
   Drupal.color = {
     logoChanged: false,
+    bannerChanged: false,
```
Comment (ronaldtebrake) [inline themes/socialblue/color/preview.js]: Looks like this isn't used anymore.

File: themes/socialblue/color/preview.js [hunk 5]
```diff
@@ -64,12 +66,28 @@
...
+          colorPreviewCover.toggleClass('cover-img cover-img-gradient');
 
+          if (colorPreviewLever.hasClass('lever-on')) {
+            localStorage.setItem('lever-on', '1');
+          }
+          else {
+            localStorage.removeItem('lever-on');
+          }
+        }
+      );і
```
Comment (ronaldtebrake) [inline themes/socialblue/color/preview.js]:
```suggestion
      );
```

**Comment by ronaldtebrake** [comment]
@agami4 Almost there, some small nitpicks, if you could fix those that would be great :) (works like a charm)!

**Comment by agami4** [comment]
@ronaldtebrake I've fixed this small nitpicks) Please check

**Comment by ronaldtebrake** [comment]
Cherry picked to 6.x as well.

**Pull request merged**